tests/bank_interface/test_client.py

```diff
@@ -116,17 +129,76 @@ def time_mock() -> Generator[Mock, None, None]:
         yield mock
 
 
-def test_bank_client_initializes(nordigen_client_mock: Mock) -> None:
+def test_bank_client_initializes_valid_access_token(
+    nordigen_client_mock: Mock, disk_token_store_mock: Mock
+) -> None:
+    disk_token_store_mock.return_value.is_access_token_valid.return_value = True
+    BankClient(
+        NordigenAuth(secret_id="mock_secret_id", secret_key="mock_secret_key"),
+        [
+            BankDetails(name="mock_name1", country="mock_country1"),
+            BankDetails(name="mock_name2", country="mock_country2"),
+        ],
+    )
+    disk_token_store_mock.return_value.get_last_token.assert_called_once_with(
+        "temporary-user-id"
+    )
+    nordigen_client_mock.assert_called_with(
+        secret_id="mock_secret_id", secret_key="mock_secret_key"
+    )
+    assert nordigen_client_mock.return_value.token == "fake-last-access-token"
+
+
+def test_bank_client_initializes_valid_refresh_token(
+    nordigen_client_mock: Mock, disk_token_store_mock: Mock
+) -> None:
+    disk_token_store_mock.return_value.is_access_token_valid.return_value = False
+    disk_token_store_mock.return_value.is_refresh_token_valid.return_value = True
+    BankClient(
+        NordigenAuth(secret_id="mock_secret_id", secret_key="mock_secret_key"),
+        [
+            BankDetails(name="mock_name1", country="mock_country1"),
+            BankDetails(name="mock_name2", country="mock_country2"),
+        ],
+    )
+    disk_token_store_mock.return_value.get_last_token.assert_called_once_with(
+        "temporary-user-id"
+    )
+    nordigen_client_mock.assert_called_with(
+        secret_id="mock_secret_id", secret_key="mock_secret_key"
+    )
+    nordigen_client_mock.return_value.exchange_token.assert_called_once_with(
+        "fake-last-refresh-token"
+    )
+
+
+def test_bank_client_initializes_invalid_token(
+    nordigen_client_mock: Mock, disk_token_store_mock: Mock
+) -> None:
+    disk_token_store_mock.return_value.is_access_token_valid.return_value = False
+    disk_token_store_mock.return_value.is_refresh_token_valid.return_value = False
+
+    nordigen_client_mock.return_value.generate_token.return_value = {
+        "access": "newly-generated-token",
+        "access_expires": 10,
+        "refresh": "newly-generated-token",
+        "refresh_expires": 50,
+    }
+
     BankClient(
         NordigenAuth(secret_id="mock_secret_id", secret_key="mock_secret_key"),
         [
             BankDetails(name="mock_name1", country="mock_country1"),
             BankDetails(name="mock_name2", country="mock_country2"),
         ],
     )
+
+    disk_token_store_mock.return_value.get_last_token.assert_not_called()
```

Comment:
may assert if the exchange token is not called as well

Reply:
`nordigen_client_mock.return_value.exchange_token` is not called here

Only `nordigen_client.mock.return_value.generate_token` is called, the assert to this is a couple of lines below